File: Offer/2286/main.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minValue: int = 1e9
        profit: int = 0
        temp: int

        for price in prices:
            temp = price - minValue
            if temp > profit:
                profit = temp
            if price < minValue:
                minValue = price
        
        return profit

File: Offer/2286/main.go
package main

func maxProfit(prices []int) int {
	if len(prices) == 0 {
		return 0
	}

	var minValue int = 1e9
	var profit, temp int = 0, 0

	for _, price := range prices {
		temp = price - minValue
		if temp > profit {
			profit = temp
		}
		if price < minValue {
			minValue = price
		}
	}
	return profit
}
